pkg/webhooks/common.go

```diff
@@ -44,7 +66,28 @@ func getEnforceFailureErrorMsg(engineResponses []*response.EngineResponse) strin
 	policyToRule := make(map[string]interface{})
 	var resourceName string
 	for _, er := range engineResponses {
-		if !er.IsSuccessful() && er.PolicyResponse.ValidationFailureAction == common.Enforce {
+		nsAction := ""
+		actionOverride := false
+
+		for _, v := range er.PolicyResponse.ValidationFailureActionOverrides {
+			action := v.Action
+			if action != common.Enforce && action != common.Audit {
+				continue
+			}
+
+			for _, ns := range v.Namespaces {
+				if wildcard.Match(ns, er.PatchedResource.GetNamespace()) {
+					nsAction = action
+					actionOverride = true
+				}
+			}
+
+			if actionOverride {
+				break
+			}
+		}
+
+		if !er.IsSuccessful() && ((actionOverride && nsAction == common.Enforce) || (!actionOverride && er.PolicyResponse.ValidationFailureAction == common.Enforce)) {
```

Comment:
Duplicate code in `isResponseSuccessful`, can we extract the logic and re-use it?

Reply:
Do you mean `toBlockResource()`?